app/src/main/kotlin/cz/covid19cz/erouska/ui/efgsagreement/EfgsAgreementVM.kt

```diff
@@ -0,0 +1,68 @@
+package cz.covid19cz.erouska.ui.efgsagreement
+
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.viewModelScope
+import arch.livedata.SafeMutableLiveData
+import com.google.android.gms.common.api.ApiException
+import cz.covid19cz.erouska.db.SharedPrefsRepository
+import cz.covid19cz.erouska.exposurenotifications.ExposureNotificationsRepository
+import cz.covid19cz.erouska.ui.base.BaseVM
+import cz.covid19cz.erouska.ui.dashboard.event.GmsApiErrorEvent
+import cz.covid19cz.erouska.ui.verification.NoKeysException
+import cz.covid19cz.erouska.ui.verification.ReportExposureException
+import cz.covid19cz.erouska.utils.L
+import kotlinx.coroutines.launch
+import java.net.UnknownHostException
+
+class EfgsAgreementVM @ViewModelInject constructor(val prefs: SharedPrefsRepository, val exposureNotificationsRepo: ExposureNotificationsRepository) :
+    BaseVM() {
+
+    val traveller = prefs.isTraveller()
+    val loading = SafeMutableLiveData(false)
+
+    fun agree() {
+        prefs.setConsentToFederation(true)
+        publishKeys()
+    }
+
+    fun disagree() {
+        prefs.setConsentToFederation(false)
+        publishKeys()
```

Comment:
why do we publish keys when the user disagrees?

Reply:
tady uživatel dává/nedává souhlas s upozorněním zahraničních appek